test/parallel/test-http-client-reject-unexpected-agent.js

```diff
@@ -0,0 +1,64 @@
+'use strict';
+const common = require('../common');
+const assert = require('assert');
+const http = require('http');
+
+const baseOptions = {
+  method: 'GET',
+  port: undefined,
+  host: '127.0.0.1',
```

Comment:
I have this change in, waiting for discussion on your other requests prior to committing that up though. :D :+1: